### ISSUE TITLE
errors when using docker compose

### ISSUE THREAD

**Issue by rastarr** [towfiqi/serpbear#12]
Tried deploying to a docker container using compose. Sadly I get the following in the logs

```
[1] Error: Settings file dont exist.
[1]     at getAppSettings (/app/cron.js:34:16)
[1] node:internal/process/promises:288
[1]             triggerUncaughtException(err, true /* fromPromise */);
[1]             ^
[1] 
[1] [Error: EACCES: permission denied, open '/app/data/settings.json'] {
[1]   errno: -13,
[1]   code: 'EACCES',
[1]   syscall: 'open',
[1]   path: '/app/data/settings.json'
[1] }
[1] 
[1] Node.js v18.12.1
[1] node cron.js exited with code 1
[0] Listening on port 3000
[0] ConnectionError [SequelizeConnectionError]: SQLITE_CANTOPEN: unable to open database file
[0]     at Database.<anonymous> (/app/node_modules/sequelize/lib/dialects/sqlite/connection-manager.js:52:25) {
[0]   parent: [Error: SQLITE_CANTOPEN: unable to open database file] {
[0]     errno: 14,
[0]     code: 'SQLITE_CANTOPEN'
[0]   },
[0]   original: [Error: SQLITE_CANTOPEN: unable to open database file] {
[0]     errno: 14,
[0]     code: 'SQLITE_CANTOPEN'
[0]   }
[0] }
```
Any ideas?

**Comment by towfiqi** [comment]
@rastarr It could be that your container cannot access the docker volume you created for the app. Did you create the volume as described in the [doc](https://docs.serpbear.com/deployment/running-locally-with-docker#using-docker-compose)?

**Comment by rastarr** [comment]
> @rastarr It could be that your container cannot access the docker volume you created for the app. Did you create the volume as described in the [doc](https://docs.serpbear.com/deployment/running-locally-with-docker#using-docker-compose)?

well the compose I've added is below

version: "3.8"

```
services:
  app:
    image: towfiqi/serpbear
    container_name: serpbear
    restart: unless-stopped
    ports:
      - 3000:3000
    environment:
      - USER=admin
      - PASSWORD=0123456789
      - SECRET=4715aed3216f7b0a38e6b534a958362654e96d10fbc04700770d572af3dce43625dd
      - APIKEY=5saedXklbslhnapihe2pihp3pih4fdnakhjwq5
      - NEXT_PUBLIC_APP_URL=http://localhost:3000
    volumes:
      - /volume1/docker/serpbear/serpbear_data:/app/data
networks:
  my-network:
    driver: bridge
volumes:
  serpbear_data:
```

So maybe something is not correct but I am a newbie at all this. Do you see anything obvious?
Appreciate your help

**Comment by towfiqi** [comment]
Can you try changing this line: `- /volume1/docker/serpbear/serpbear_data:/app/data` to this:   `- serpbear_data:/app/data`

**Comment by rastarr** [comment]
> Can you try changing this line: `- /volume1/docker/serpbear/serpbear_data:/app/data` to this: `- serpbear_data:/app/data`

Nope - Failure X

failed to deploy a stack:
service "app" refers to
undefined volume
serpbear_data: invalid compose
project

**Comment by towfiqi** [comment]
Sounds like docker compose cannot create the volume. 
Can you try running  `docker volume create serpbear_data` and then run `docker compose up`

**Comment by towfiqi** [comment]
Great! :)

**Comment by morvy** [comment]
I know this is a resolved issue, but I think there is an issue with running SerpBear on standard mounted paths. If I create a volume and assign it, it works fine, but as soon as I assign a path (existing), SerpBear cannot access settings.json or anything it needs to create. I tried to set PUID / PGID to user owning the directory (1026:100 as first user in Synology) but nothing changed. I tried to run it on multiple machines with Docker and couldn't start it without docker volume.
